delete_me_proxies_example/wqp_discrete_inventory.R

```diff
@@ -0,0 +1,655 @@
+                                              yes = "particulate",
+                                              no = ifelse(is.na(ResultSampleFractionText) | ResultSampleFractionText %in% 
+                                                            c("NA", "None", "Pot. Dissolved", "Filterable", "Non-filterable", "Non-Filterable (Particle)"),
+                                                          yes = "unknown",
+                                                          no = ResultSampleFractionText))))) %>%
+      # Create FLAG if fraction is unknown (thus ambiguous)
+      mutate(flag_fracAmbiguous = ResultSampleFractionText == "unknown") %>% 
+      
+      # Flag records w/ zero values and not enough information to resolve <-- meaning ResultMeasureValue was zero but DetectionQuantitationLimitMeasure.MeasureValue was also zero or missing. 
+      mutate(flag_resultZero = flag_missingResult | 
+               ((ResultMeasureValue == 0 & is.na(DetectionQuantitationLimitMeasure.MeasureValue)) | 
+                  (is.na(ResultMeasureValue) & DetectionQuantitationLimitMeasure.MeasureValue == 0) | 
+                  (ResultMeasureValue == 0 & DetectionQuantitationLimitMeasure.MeasureValue == 0))) %>% 
+      
+      # Update resultVal, resultUnits, and remarkCode for non-detects
+      update_cols_for_non_detects() %>% 
+      
+      # Now remove missing data & 0s
+      filter(!flag_resultZero) %>% 
+      
+      # Filter data based on Proxy group specific needs:
+      filter_habs_data(characteristic_df) %>%
+      filter_pfas_data(characteristic_df) %>%
+      filter_metals_data(characteristic_df) %>%
+      # Keep "fraction" separate for metals, but not the others
+      mutate(Fraction = ifelse(CharacteristicName %in% 
+                                 pull(filter(characteristic_df, Proxies_Grp == "METALS"), CharacteristicName), 
+                               ResultSampleFractionText, "ALL")) %>% 
+      
+      # Remove duplicate data (Jenny was using "collectingOrganization", but that wasn't in this data)
+      group_by(OrganizationIdentifier, MonitoringLocationIdentifier, ActivityStartDate, ActivityStartTime.Time, CharacteristicName, ResultSampleFractionText) %>% 
+      mutate(n_duplicated = n()) %>% 
+      filter(n_duplicated == 1) %>%
+      select(-n_duplicated) %>%
+      ungroup() 
+    
+    write_feather(basin_data_clean, basin_cleaned_fn)
+    
+  } else {
+    basin_data_clean <- read_feather(basin_cleaned_fn)
+  }
+  
+  # How much data got filtered out? 97% for DRB?!
+  message(sprintf("During cleaning process, %s%% of the %s data was filtered out. %s observations remain.", 
+                  round((nrow(basin_data) - nrow(basin_data_clean)) / nrow(basin_data) * 100, 2), 
+                  basin_nm, prettyNum(nrow(basin_data_clean), ",")))
+  
+  return(basin_cleaned_fn)
+}
+
+add_site_metadata <- function(basin_data_fn, basin_site_fn) {
```

Comment:
As a whole, these scripts do a great job naming things. We'll want to be consistent in the use of camel case or snake case (I'm a devotee of `snake_case_personally` 😄).

Reply:
#team_snake_case 😺

Reply:
I am used to be a camel case kinda gal, but am now almost entirely in the 🐍 case camp 😆